components/script/dom/window.rs

```diff
@@ -1790,13 +1791,20 @@ impl Window {
                 // then put it in the delaying load events mode.
                 self.window_proxy().start_delaying_load_events_mode();
             }
-            self.main_thread_script_chan()
-                .send(MainThreadScriptMsg::Navigate(
-                    pipeline_id,
-                    LoadData::new(url, Some(pipeline_id), Some(referrer), referrer_policy),
-                    replace,
-                ))
-                .unwrap();
+            // TODO: step 11, navigationType.
+            // Step 12, 13
+            let load_data = if let Some(mut load_data) = load_data {
```

Comment:
Should we just replace the parameters of this function with a single `load_data`? Is there a reason to have both?

Reply:
Yes that's a good idea...